base/game_events/PsynergyStoneEvent.ts

```diff
@@ -0,0 +1,154 @@
+import {CharsStatusWindow} from "../windows/CharsStatusWindow";
+import {NPC} from "../NPC";
+import {base_actions, directions} from "../utils";
+import {DialogManager} from "../utils/DialogManager";
+import {GameEvent, event_types} from "./GameEvent";
+import {Button} from "../XGamepad";
+import * as _ from "lodash";
+
+const INIT_TEXT = (hero_name: string) => `${hero_name} checked on the ground...`;
+const GET_TEXT = (hero_name: string) => `${hero_name} got a Psynergy Stone.`;
+const RECOVERY_TEXT = (hero_name: string) => `${hero_name}'s PP are maxed out!`;
+const DISAPPEAR_TEXT = `The Psynergy Stone disappeared...`;
+
+export class PsynergyStoneEvent extends GameEvent {
+    private control_enable: boolean = true;
+    private running: boolean = false;
+    private resolve: Function;
+    private promise: Promise<any>;
+    private dialog_manager: DialogManager;
+    private custom_init_text: string;
```

Comment:
In this case, I think you can remove `custom_init_text` property, for this situation, it won't change at all.